src/main/java/mops/presentation/FileController.java

```diff
@@ -0,0 +1,55 @@
+package mops.presentation;
+
+import lombok.AllArgsConstructor;
+import mops.businesslogic.Account;
+import mops.businesslogic.FileService;
+import mops.businesslogic.utils.AccountUtil;
+import org.keycloak.adapters.springsecurity.token.KeycloakAuthenticationToken;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("material1/file")
+@AllArgsConstructor
+public class FileController {
+
+    /**
+     * Handles actions on single files: Delete, Preview, Download.
+     */
+    private FileService fileService;
+
+    /**
+     * @param token   a keycloak authentication token
+     * @param model   spring boot view model
+     * @param fileId the id of the requested file
+     * @return the route to template 'file'
+     */
+    @GetMapping("/{fileId}")
+    public String getFile(KeycloakAuthenticationToken token,
+                                         Model model,
+                                         @PathVariable("fileId") long fileId) {
+        Account account = AccountUtil.getAccountFromToken(token);
+        fileService.getFile(account, fileId); //TODO: Replace int with file type (tbd)
+        //TODO: Implement logic for actual file display
+        return "file"; //TODO: route to file display
```

Comment:
I think we can already implement it, because we only want to display the file and this is possible without business logic as we can mock it in tests.

Reply:
Yes - however we don't have a File class yet - or should we just use file info in that case?

Reply:
We have a `FileInfo` class which contains basic info about the file